folly/CMakeLists.txt

```diff
@@ -13,3 +13,31 @@ install(
 )
 
 add_subdirectory(experimental/exception_tracer)
+
+get_target_property(pic folly POSITION_INDEPENDENT_CODE)
+if (pic AND CYTHON_FOUND AND PYTHONINTERP_FOUND)
+  # Create tree of symbolic links in structure required for successful
+  # compliation by Cython.
+  #   - must be in path named same as extension
+
+  set(_cybld "${CMAKE_CURRENT_BINARY_DIR}/cybld")
+
+  file(MAKE_DIRECTORY "${_cybld}/folly")
+  foreach(_src
+    "executor.pxd"
+    "executor.pyx"
+    "futures.pxd"
+    "__init__.pxd"
+    "AsyncioExecutor.h")
+    message(STATUS "Linking ${CMAKE_CURRENT_SOURCE_DIR}/python/${_src} to ${_cybld}/folly/${_src}")
+    add_custom_target("create_folly_link_${_src}" ALL
+    COMMAND ${CMAKE_COMMAND} -E create_symlink ${CMAKE_CURRENT_SOURCE_DIR}/python/${_src} "${_cybld}/folly/${_src}")
+  endforeach()
+
+  # Install Folly Python Bindings
+  install(CODE "
```

Comment:
I guess this has to be done only at install-time since we don't know DESTDIR until then?
Is it possible to invoke setup.py build_ext during the build phase without DESTDIR, and then only do installation during the install phase?

Reply:
Code updated with a separate build and install phase.
I note however, that even without change the compile step is re-run on the second pass.